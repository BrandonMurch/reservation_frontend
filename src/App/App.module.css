.app {
    height: 100vh;
    width: 100vh;
    /* overflow: hidden; */
}

.textContainer {
    composes: centerVertically from '../CSS/helpers.module.css';
    width: 100vw;
    top: 50vh;
    text-align: center;
    cursor: pointer;
    opacity: 1;
}

.textContainerTransparent {
    composes: textContainer;
    opacity: 0.1;

}

.text {
    font-size: 3rem;
}
