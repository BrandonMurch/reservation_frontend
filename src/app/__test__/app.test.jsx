import React from 'react';
import {
  render, fireEvent, getByTestId,
} from '@testing-library/react';
import { unmountComponentAtNode } from 'react-dom';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import App from '../index';

describe('<App />', () => {
  let appContainer = null;
  beforeEach(() => {
    appContainer = document.createElement('div');
    document.body.appendChild(appContainer);
  });

  afterEach(() => {
    unmountComponentAtNode(appContainer);
    appContainer.remove();
    appContainer = null;
  });

  it('should match snapshot', () => {
    const component = renderer.create(<App />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should display text', () => {
    const { getByText } = render(<App />);
    const linkElement = getByText('Click here to make a reservation');
    expect(linkElement).toBeInTheDocument();
  });
  it('should open overlay window when link is clicked', () => {
    const { container } = render(<App />);
    const calendarLink = getByTestId(container, 'opening-link-calendar');
    fireEvent.click(calendarLink);
    const overlayWindow = getByTestId(container, 'overlay-window');
    expect(overlayWindow).toBeInTheDocument();
  });
});
