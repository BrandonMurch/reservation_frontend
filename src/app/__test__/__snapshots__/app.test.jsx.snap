// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match snapshot 1`] = `
<div
  className="textContainer"
>
  <a
    className="text"
    data-testid="opening-link-calendar"
    href="/calendar"
    onClick={[Function]}
  >
    Click here to make a reservation
  </a>
</div>
`;
